src/csv/index.js

```diff
@@ -233,82 +251,93 @@ class TransactionParser extends Transform {
   }
 
   _transform(row, encoding, callback) {
-    this.lineOffset++;
+    try {
```

Comment:
placing a try block around all this, and refactoring it to use the above new normalization functions. additionally, some changes in the `if (!adapter)` to be a little easier to follow and to always skip the header row itself